src/ui/MainMenu.cpp

```diff
@@ -8730,46 +9967,536 @@ namespace MainMenu {
+		        {average, average, average, average, average, average}, // shaman
+		        {bad, good, bad, average, good, average},               // hunter
+		    };
+
+		    // difficulty star ratings
+		    static constexpr int difficulty[][2] = {
+		        {3, 1}, // barbarian
+		        {4, 2}, // warrior
+		        {4, 2}, // healer
+		        {2, 2}, // rogue
+		        {4, 1}, // wanderer
+		        {3, 2}, // cleric
+		        {3, 2}, // merchant
+		        {3, 2}, // wizard
+		        {2, 2}, // arcanist
+		        {1, 3}, // joker
+		        {3, 4}, // sexton
+		        {2, 2}, // ninja
+		        {5, 1}, // monk
+		        {3, 3}, // conjurer
+		        {1, 3}, // accursed
+		        {3, 3}, // mesmer
+		        {2, 5}, // brewer
+		        {2, 5}, // mechanist
+		        {2, 4}, // punisher
+		        {2, 4}, // shaman
+		        {2, 2}, // hunter
+		    };
+
+		    for (int c = 0; c < num_class_stats; ++c) {
+		        static auto class_stat_fn = [](Field& field, int index){
+			        const int i = std::min(std::max(0, client_classes[index]), num_class_descs - 1);
+			        const int s = (int)strtol(field.getName(), nullptr, 10);
+			        field.setColor(class_stat_colors[i][s]);
+		        };
+		        static char buf[16];
+		        snprintf(buf, sizeof(buf), "%d", c);
+		        auto class_stat = card->addField(buf, 16);
+		        class_stat->setSize(SDL_Rect{
+		            bottom.x + column * c, bottom.y, column, bottom.h});
+		        class_stat->setHJustify(Field::justify_t::CENTER);
+		        class_stat->setVJustify(Field::justify_t::TOP);
+		        class_stat->setFont(smallfont_outline);
+		        class_stat->setText(class_stats_text[c]);
+		        switch (index) {
+		        case 0: class_stat->setTickCallback([](Widget& widget){class_stat_fn(*static_cast<Field*>(&widget), 0);}); break;
+		        case 1: class_stat->setTickCallback([](Widget& widget){class_stat_fn(*static_cast<Field*>(&widget), 1);}); break;
+		        case 2: class_stat->setTickCallback([](Widget& widget){class_stat_fn(*static_cast<Field*>(&widget), 2);}); break;
+		        case 3: class_stat->setTickCallback([](Widget& widget){class_stat_fn(*static_cast<Field*>(&widget), 3);}); break;
+		        }
+		        (*class_stat->getTickCallback())(*class_stat);
+		    }
+
+		    // difficulty header
+		    constexpr SDL_Rect difficulty_size{76, 306, 172, 64};
+		    auto difficulty_header = card->addField("difficulty_header", 128);
+		    difficulty_header->setFont(smallfont_outline);
+		    difficulty_header->setColor(makeColorRGB(209, 166, 161));
+		    difficulty_header->setText("Survival\nComplexity");
+		    difficulty_header->setHJustify(Field::justify_t::LEFT);
+		    difficulty_header->setVJustify(Field::justify_t::BOTTOM);
+		    difficulty_header->setSize(difficulty_size);
+
+		    // difficulty stars
+		    static constexpr int star_buf_size = 32;
+	        static auto stars_fn = [](Field& field, int index){
```

Comment:
My compiler asking to pass variables into capture list, so needs `static auto stars_fn = [bad, good, poor, decent, average](Field& field, int index){`

Reply:
don't use capture list, make bad/good/poor/decent/average static. also upgrade your compiler!